### ISSUE TITLE
Training restarts instead of resuming from the last checkpoint

### ISSUE THREAD

**Issue by JustineBoutry** [DeepLabCut/DeepLabCut#2743]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

I encountered an issue where DeepLabCut restarts training from **iteration 100** instead of resuming from the last saved checkpoint. Even though I had previously trained the network up to **55,000 iterations**, when I attempted to resume training after an interruption, it reloaded the **pre-trained weights** (ImageNet) and started again from iteration 100.

### Operating System

I'm on google collab, I tried all the different GPU

### DeepLabCut version

DLC 2.3.10

### DeepLabCut mode

multi animal

### Device type

I tried the different ones on google collab+ and it doesn't change this issue

### Steps To Reproduce

1. Train a model for several iterations (e.g., up to 55,000 iterations) with google collab
2. Stop the training and attempt to resume it using `train_network`.
3. Instead of resuming from the last checkpoint, the training starts from iteration 100 with the ImageNet-pretrained weights.

### Relevant log output

```shell
Selecting multi-animal trainer
Config:
{'all_joints': [[0], [1], [2], [3], [4], [5], [6], [7]],
 'all_joints_names': ['tip_head',
                      'base_tail',
                      'tip_tail',
                      'mouth',
                      'l_antena_base',
                      'r_antena_base',
                      'l_antena_tip',
                      'r_antena_tip'],
 'alpha_r': 0.02,
 'apply_prob': 0.5,
 'batch_size': 8,
 'contrast': {'clahe': True,
              'claheratio': 0.1,
              'histeq': True,
              'histeqratio': 0.1},
 'convolution': {'edge': False,
                 'emboss': {'alpha': [0.0, 1.0], 'strength': [0.5, 1.5]},
                 'embossratio': 0.1,
                 'sharpen': False,
                 'sharpenratio': 0.3},
 'crop_pad': 0,
 'crop_sampling': 'hybrid',
 'crop_size': [400, 400],
 'cropratio': 0.4,
 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_DaphniaSwimmingSep4/DaphniaSwimming_boutry95shuffle1.pickle',
 'dataset_type': 'multi-animal-imgaug',
 'decay_steps': 30000,
 'deterministic': False,
 'display_iters': 100,
 'fg_fraction': 0.25,
 'global_scale': 0.8,
 'init_weights': '/usr/local/lib/python3.10/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt',
 'intermediate_supervision': False,
 'intermediate_supervision_layer': 12,
 'location_refinement': True,
 'locref_huber_loss': True,
 'locref_loss_weight': 0.05,
 'locref_stdev': 7.2801,
 'log_dir': 'log',
 'lr_init': 0.0005,
 'max_input_size': 1500,
 'max_shift': 0.4,
 'mean_pixel': [123.68, 116.779, 103.939],
 'metadataset': 'training-datasets/iteration-0/UnaugmentedDataSet_DaphniaSwimmingSep4/Documentation_data-DaphniaSwimming_95shuffle1.pickle',
 'min_input_size': 64,
 'mirror': False,
 'multi_stage': True,
 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]],
 'net_type': 'resnet_50',
 'num_idchannel': 0,
 'num_joints': 8,
 'num_limbs': 28,
 'optimizer': 'adam',
 'pafwidth': 20,
 'pairwise_huber_loss': False,
 'pairwise_loss_weight': 0.1,
 'pairwise_predict': False,
 'partaffinityfield_graph': [[0, 1],
                             [0, 2],
                             [0, 3],
                             [0, 4],
                             [0, 5],
                             [0, 6],
                             [0, 7],
                             [1, 2],
                             [1, 3],
                             [1, 4],
                             [1, 5],
                             [1, 6],
                             [1, 7],
                             [2, 3],
                             [2, 4],
                             [2, 5],
                             [2, 6],
                             [2, 7],
                             [3, 4],
                             [3, 5],
                             [3, 6],
                             [3, 7],
                             [4, 5],
                             [4, 6],
                             [4, 7],
                             [5, 6],
                             [5, 7],
                             [6, 7]],
 'partaffinityfield_predict': True,
 'pos_dist_thresh': 17,
 'pre_resize': [],
 'project_path': '/content/drive/MyDrive/Post-doct/VideoDaphnia/DaphniaSwimming-boutry-2024-09-04',
 'regularize': False,
 'rotation': 25,
 'rotratio': 0.4,
 'save_iters': 1000,
 'scale_jitter_lo': 0.5,
 'scale_jitter_up': 1.25,
 'scoremap_dir': 'test',
 'shuffle': True,
 'snapshot_prefix': '/content/drive/MyDrive/Post-doct/VideoDaphnia/DaphniaSwimming-boutry-2024-09-04/dlc-models/iteration-0/DaphniaSwimmingSep4-trainset95shuffle1/train/snapshot',
 'stride': 8.0,
 'weigh_negatives': False,
 'weigh_only_present_joints': False,
 'weigh_part_predictions': False,
 'weight_decay': 0.0001}
Activating limb prediction...
Batch Size is 8
Getting specs multi-animal-imgaug 28 8
/usr/local/lib/python3.10/dist-packages/tensorflow/python/keras/engine/base_layer_v1.py:1694: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  warnings.warn('`layer.apply` is deprecated and '
/usr/local/lib/python3.10/dist-packages/tf_slim/layers/layers.py:684: UserWarning: `layer.apply` is deprecated and will be removed in a future version. Please use `layer.__call__` method instead.
  outputs = layer.apply(inputs, training=is_training)
Loading ImageNet-pretrained resnet_50
Downloading a ImageNet-pretrained model from http://download.tensorflow.org/models/resnet_v1_50_2016_08_28.tar.gz....
Max_iters overwritten as 100000
Display_iters overwritten as 100
Save_iters overwritten as 1000
Training parameters:
{'stride': 8.0, 'weigh_part_predictions': False, 'weigh_negatives': False, 'fg_fraction': 0.25, 'mean_pixel': [123.68, 116.779, 103.939], 'shuffle': True, 'snapshot_prefix': '/content/drive/MyDrive/Post-doct/VideoDaphnia/DaphniaSwimming-boutry-2024-09-04/dlc-models/iteration-0/DaphniaSwimmingSep4-trainset95shuffle1/train/snapshot', 'log_dir': 'log', 'global_scale': 0.8, 'location_refinement': True, 'locref_stdev': 7.2801, 'locref_loss_weight': 0.05, 'locref_huber_loss': True, 'optimizer': 'adam', 'intermediate_supervision': False, 'intermediate_supervision_layer': 12, 'regularize': False, 'weight_decay': 0.0001, 'crop_pad': 0, 'scoremap_dir': 'test', 'batch_size': 8, 'dataset_type': 'multi-animal-imgaug', 'deterministic': False, 'mirror': False, 'pairwise_huber_loss': False, 'weigh_only_present_joints': False, 'partaffinityfield_predict': True, 'pairwise_predict': True, 'all_joints': [[0], [1], [2], [3], [4], [5], [6], [7]], 'all_joints_names': ['tip_head', 'base_tail', 'tip_tail', 'mouth', 'l_antena_base', 'r_antena_base', 'l_antena_tip', 'r_antena_tip'], 'alpha_r': 0.02, 'apply_prob': 0.5, 'contrast': {'clahe': True, 'claheratio': 0.1, 'histeq': True, 'histeqratio': 0.1, 'gamma': False, 'sigmoid': False, 'log': False, 'linear': False}, 'convolution': {'edge': False, 'emboss': {'alpha': [0.0, 1.0], 'strength': [0.5, 1.5]}, 'embossratio': 0.1, 'sharpen': False, 'sharpenratio': 0.3}, 'crop_sampling': 'hybrid', 'crop_size': [400, 400], 'cropratio': 0.4, 'dataset': 'training-datasets/iteration-0/UnaugmentedDataSet_DaphniaSwimmingSep4/DaphniaSwimming_boutry95shuffle1.pickle', 'decay_steps': 30000, 'display_iters': 100, 'init_weights': '/usr/local/lib/python3.10/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt', 'lr_init': 0.0005, 'max_input_size': 1500, 'max_shift': 0.4, 'metadataset': 'training-datasets/iteration-0/UnaugmentedDataSet_DaphniaSwimmingSep4/Documentation_data-DaphniaSwimming_95shuffle1.pickle', 'min_input_size': 64, 'multi_stage': True, 'multi_step': [[0.0001, 7500], [5e-05, 12000], [1e-05, 200000]], 'net_type': 'resnet_50', 'num_idchannel': 0, 'num_joints': 8, 'num_limbs': 28, 'pafwidth': 20, 'pairwise_loss_weight': 0.1, 'partaffinityfield_graph': [[0, 1], [0, 2], [0, 3], [0, 4], [0, 5], [0, 6], [0, 7], [1, 2], [1, 3], [1, 4], [1, 5], [1, 6], [1, 7], [2, 3], [2, 4], [2, 5], [2, 6], [2, 7], [3, 4], [3, 5], [3, 6], [3, 7], [4, 5], [4, 6], [4, 7], [5, 6], [5, 7], [6, 7]], 'pos_dist_thresh': 17, 'pre_resize': [], 'project_path': '/content/drive/MyDrive/Post-doct/VideoDaphnia/DaphniaSwimming-boutry-2024-09-04', 'rotation': 25, 'rotratio': 0.4, 'save_iters': 1000, 'scale_jitter_lo': 0.5, 'scale_jitter_up': 1.25, 'pseudo_threshold': 0.1, 'video_path': '', 'traintime_resize': False}
Starting multi-animal training....
iteration: 100 loss: 0.0551 scmap loss: 0.0395 locref loss: 0.0032 limb loss: 0.0124 lr: 0.0001
iteration: 200 loss: 0.0079 scmap loss: 0.0067 locref loss: 0.0002 limb loss: 0.0009 lr: 0.0001
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by JustineBoutry** [comment]
Oh! And of course, I'm a beginner, so I likely did something wrong. Any advice would be amazing. 

Best, 
Justine

**Comment by n-poulsen** [comment]
Hi @JustineBoutry! You should be able to resume training from the existing checkpoint by updating the `'init_weights': '/usr/local/lib/python3.10/dist-packages/deeplabcut/pose_estimation_tensorflow/models/pretrained/resnet_v1_50.ckpt'` value in the `pose_cfg.yaml` for the shuffle you're working on.

In the folder for your DeepLabCut project on Google COLAB, you'll have a folder for your shuffle which should look like this:

![pose-cfg](https://github.com/user-attachments/assets/7e9341e5-1078-4a56-9f76-2af2a9827962)

Here, to restart training from an existing snapshot (e.g. 55,000 iterations), set the `init_weights` to the path to your snapshot-55000 model. You get the path by clicking on the "..." next to the file and selecting "Copy Path" It should be something like:

```
'init_weights': /content/drive/MyDrive/.../my-deeplabcut-project-2024-05-20/dlc-models/iteration-0/mouseMay20-trainset95shuffle1/train/snapshot-100
```

Make sure the filename is `snapshot-XXX` (and not `snapshot-100.index` - just the iterations).

Let me know if that fixes your issue!
